### ISSUE TITLE
[FEATURE] Add support for a JSON file for storing configuration options instead of only using CLI

### ISSUE THREAD

**Issue by Chowarmaan** [compodoc/compodoc#379]
##### **Overview of the issue**
The command line is getting rather large and unwieldly in the package.json for a complex build of documentation.  It would be nice to be able to set the configuration using a compodoc.json file (or whatever name specified) similar to the tsconfig.json. 

##### **Operating System, Node.js, npm, compodoc version(s)**
Windows 10
Node 8.8.1
npm: 5.5.1
Compodoc: 1.0.4

##### **Motivation for or Use Case**
Cleanliness of the package.json to have the configuration in a structured json file like a number of other tools, especially as the options continue to grow on the application.

##### **Suggest a Fix**
Add an additional option to specify the json configuration file and possibly even default to compodoc.json if a file is not provided or found.

**Comment by vogloblinsky** [comment]
Yes Steven, this is something we have to do. I had it for 1.1.0 milestone.

**Comment by jensschneider** [comment]
That would be a great thing. Too bad that it will be pushed from version to version.
